File: Length of Longest Subarray With at Most K Frequency.cpp
class Solution 
{
public:
    int maxSubarrayLength(vector<int>& nums, int k) 
    {
        int n = nums.size();
        unordered_map<int, int>freq;
        int start = -1, end = 0;
        int res = INT_MIN;

        for(int i = 0; i < n; i++)
        {
            freq[nums[i]]++;
            while(freq[nums[i]] > k)
            {
                start++;
                freq[nums[start]]--;
            }
            if(end - start > res)
            {
                res = end - start;
            }
            end++;
        }
        return res;
    }
};

File: Length of Longest Subarray With at Most K Frequency.py
class Solution(object):
    def maxSubarrayLength(self, nums, k):
        
        freq = Counter()
        start = -1
        end = 0
        res = 0

        for i in range(len(nums)):
            freq[nums[i]] += 1
            while freq[nums[i]] > k:
                start += 1
                freq[nums[start]] -= 1

            res = max(res, end - start)
            end +=1
            
        return res
        